hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/source/IncrementalInputSplits.java

```diff
@@ -302,6 +311,32 @@ private Stream<HoodieInstant> maySkipCompaction(Stream<HoodieInstant> instants)
         : instants;
   }
 
+  private Stream<FileSlice> filterFileSliceWithValidFiles(FileSystem fs, Stream<FileSlice> fileSlices) {
+    return fileSlices.filter(fileSlice -> {
+      Option<String> basePath = fileSlice.getBaseFile().map(BaseFile::getPath);
+      try {
+        if (!basePath.isPresent()) {
+          return true;
+        }
+        return fs.exists(new org.apache.hadoop.fs.Path(basePath.get()));
+      } catch (IOException e) {
+        LOG.error("Checking exists of base path: {} error", basePath);
+        throw new HoodieException(e);
+      }
```

Comment:
The logic is still incorrect, when a file slice has base file that does not exist but the log files exist, we should keep it.
In other words, we should keep the file slice if any base/log file exists :)

Reply:
Yes, I get it.